lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -79,6 +94,17 @@ static std::string getDiagKindString(DiagnosticKind Kind) {
   llvm_unreachable("Unhandled DiagKind in switch.");
 }
 
+/// Render the verifier syntax for a given set of educational notes.
+static std::string renderEducationalNotes(llvm::StringSet<> &EducationalNotes) {
```

Comment:
You cannot use a StringSet here without some kind of intervening sort. This will lead to non-determinism in the output when different platforms or build configurations inevitably shake up the order of iteration.

Reply:
🤦‍♂ Switched the captured diags to just use a vector, 95% of the time they'll only be one item so dealing with sets doesn't seem worth it.